card/mod/05_standard/set/rstar/rules.rb

```diff
@@ -217,17 +267,106 @@ def find_current_rule_card
     # self.card is a POTENTIAL rule; it quacks like a rule but may or may not exist.
     # This generates a prototypical member of the POTENTIAL rule's set
     # and returns that member's ACTUAL rule for the POTENTIAL rule's setting
-    set_prototype = card.trunk.prototype
-    rule_card = if card.new_card?
-      setting = card.right and set_prototype.rule_card setting.codename   
+    if card.new_card?
+       ((setting = card.right) && card.set_prototype.rule_card(setting.codename, :user=>card.rule_user)) ||
+            Card.new(:name=> "#{Card[:all].name}+#{card.rule_user_setting_name}")
```

Comment:
Ok, I get it now.  It used to return the prototype, but now you can get that from the card if you need it.  Just returns the rule card.  Good stuff.